test/nodetool/test_compact.py

```diff
@@ -67,7 +67,7 @@ def test_token_range_compatibility_argument(nodetool):
     nodetool("compact", "system_schema", "--start-token", "0", "--end-token", "1000", expected_requests=dummy_request)
 
 
-def test_partition_compatibility_argument(nodetool):
+def test_partition_compatibility_argument(nodetool, scylla_only):
```

Comment:
I added it here because our documentation mentions it. This is yet another one of those options that we silently swallow, because it was either never implemented in our backend or it is just incompatible with scylla and we'll never implement it. There will be much more of these options and I'm thorn on how to best handle them.

Reply:
Maybe we should just remove this from our docs. Our current nodetool doesn't accept it anyway.

Reply:
Yes, I think we should remove it from the docs **and** more importantly, from our tool. What good could "silently swallowing" a major option do? Supposedly somebody who ran "nodetool compact --partition 0" only expects (?!) to compact partition 0 (why?), and if we instead compact everything, the user should know that it's not supported...

Reply:
docs PR: https://github.com/scylladb/scylladb/pull/15626

Reply:
> Supposedly somebody who ran "nodetool compact --partition 0" only expects (?!) to compact partition 0 (why?)

What currently happens is that nodetool interprets `--partition` and `0` as keyspace and table names and sends them to Scylla, which will then send an ugly error that this keyspace/table is not found.

Reply:
So shouldn't our tool do the same? It sounds to me we should remove this from the tool, not just from the documentation (of course, we don't need to do this now in this PR).

By the way, CQL only allows keyspace and table names to contain alphanumeric characters and underscores - "--partition" isn't even valid keyspace name. But since the DynamoDB API does allow a dash in a table name, I think we should support it. Maybe we need - like most Unix tools - a "--" option after which any name, even if it starts with -, is treated as a keyspace name and not an option.

Reply:
I removed it from scylla-nodetool as well, see the v4 changelog.